pkg/services/cloudmigration/cloudmigrationimpl/xorm_store.go

```diff
@@ -433,3 +448,45 @@ func (ss *sqlStore) decryptToken(ctx context.Context, cm *cloudmigration.CloudMi
 
 	return nil
 }
+
+func (ss *sqlStore) DeleteMigrationSessionWithRelatedElements(ctx context.Context, sessionUID string) (*cloudmigration.CloudMigrationSession, error) {
+	// first we try to delete all the associated information to the session
```

Comment:
All of the database transactions need to be wrapped in an `InTransaction` block, e.g. [here](https://github.com/grafana/grafana/blob/eb72c3f8d3e83e04620ec524de3a8331a462ac73/pkg/services/cloudmigration/cloudmigrationimpl/xorm_store.go#L317) to ensure atomicity